subsys/bluetooth/host/iso.c

```diff
@@ -1934,8 +1939,11 @@ void hci_le_big_complete(struct net_buf *buf)
 
 	i = 0;
 	SYS_SLIST_FOR_EACH_CONTAINER(&big->bis_channels, bis, node) {
-		bis->iso->handle = sys_le16_to_cpu(evt->handle[i++]);
-		bt_conn_set_state(bis->iso, BT_CONN_CONNECTED);
+		struct bt_conn *iso_conn = bis->iso;
+
+		iso_conn->iso.seq_num = 0;
+		iso_conn->handle = sys_le16_to_cpu(evt->handle[i++]);
```

Comment:
It's a bad idea to include operations in the endianness macro parameter. For little endian architectures, the expression is executed more than once, and the consequence in this case is double increments and a random handle.